persist-mongodb/src/main/kotlin/io/saagie/updatarium/persist/MongodbPersistEngine.kt

```diff
@@ -36,8 +37,12 @@ const val COLLECTION = "changeset"
 
 class MongodbPersistEngine(override val configuration: PersistConfig = PersistConfig()) : PersistEngine(configuration) {
 
+    private val client: MongoClient by lazy {
+        KMongo.createClient(ConnectionString(getConnectionString()))
+    }
+
     private val collection by lazy {
-        with(KMongo.createClient(ConnectionString(getConnectionString()))) {
+        with(client) {
```

Comment:
If `client` return a `Closeable` instance, we can avoid to close it after and replace the `when` with a `use` 


Any idea to have that ?